lib/rules/no-deprecated-delete-set.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * @author Wayne Zhang
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+const utils = require('../utils')
+const { findVariable } = require('@eslint-community/eslint-utils')
+
+const DeprecatedApis = ['set', 'delete']
+const DeprecatedDollarApis = new Set(DeprecatedApis.map((item) => `$${item}`))
+
+/**
+ * @param {Expression|Super} node
+ */
+function isVue(node) {
+  return node.type === 'Identifier' && node.name === 'Vue'
+}
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'disallow using deprecated `$delete` and `$set` (in Vue.js 3.0.0+)',
+      categories: ['vue3-essential'],
```

Comment:
Please don't add the rule to any categories yet; that would be a breaking change. I'll note it down for our next major version.